glean-core/android/src/test/java/mozilla/telemetry/glean/net/BaseUploaderTest.kt

```diff
@@ -95,4 +96,43 @@ class BaseUploaderTest {
             headersCaptor.value.find { it.first == "X-Debug-ID" }!!.second
         )
     }
+
+    @Test
+    fun `splitPingForLog() correctly divides large pings`() {
+        // This makes 10 full chunks plus a chunk with 10 bytes (1 message) to test breaking out
+        // a smaller last chunk
+        val testString = "*TestData*"
+        val testData = Strings.repeat(testString, BaseUploader.MAX_LOG_PAYLOAD_SIZE_BYTES + 1)
+        val testPath = "test/Glean"
+
+        val chunks = BaseUploader.splitPingForLog(testData, testPath)
+
+        assertEquals("Must have correct number of chunks", 11, chunks.size)
+        var curChunk = 0
+        for (chunk in chunks) {
+            val calculatedHeaderMsg =
+                "Glean ping to URL: $testPath [Part ${curChunk + 1} of ${chunks.size}]"
+            assertEquals(
+                "First line must contain correct header message",
+                calculatedHeaderMsg,
+                chunk.lines()[0]
+            )
+
+            if (chunk === chunks.last()) {
```

Comment:
Can we please also check the content of each chunk?

Reply:
Sure!